File: libgomp/testsuite/libgomp.c/cancel-parallel-1.c
/* { dg-do run } */
/* { dg-set-target-env-var OMP_CANCELLATION "true" } */

#include <stdlib.h>
#include <omp.h>

int
main ()
{
  #pragma omp parallel num_threads (32)
  {
    #pragma omp cancel parallel
    if (omp_get_cancellation ())
      abort ();
  }
  return 0;
}

File: libgomp/testsuite/libgomp.fortran/cancel-parallel-1.f90
! { dg-do run }
! { dg-set-target-env-var OMP_CANCELLATION "true" }

  use omp_lib

  !$omp parallel num_threads(32)
    !$omp cancel parallel
    if (omp_get_cancellation ()) STOP 1
  !$omp end parallel
end
